pkg/pillar/cmd/zedagent/localinfo.go

```diff
@@ -0,0 +1,139 @@
+	"github.com/lf-edge/eve/pkg/pillar/zedcloud"
+)
+
+const (
+	localInfoURLPath      = "/api/v1/info"
+	localInfoPOSTInterval = time.Minute
+)
+
+func updateLocalInfoTicker(ctx *getconfigContext, multiplier float64) {
+	max := multiplier * float64(localInfoPOSTInterval)
+	min := max * 0.3
+	ctx.localInfoPOSTTicker.UpdateRangeTicker(time.Duration(min), time.Duration(max))
+}
+
+func initializeLocalInfo(ctx *getconfigContext) {
+	max := float64(localInfoPOSTInterval)
+	min := max * 0.3
+	ctx.localInfoPOSTTicker = flextimer.NewRangeTicker(time.Duration(min), time.Duration(max))
+}
+
+func triggerLocalInfoPOST(ctx *getconfigContext) {
+	log.Functionf("Triggering POST for %s to local server", localInfoURLPath)
+	ctx.localInfoPOSTTicker.TickNow()
+}
+
+// Run a periodic POST request to send information message to local server.
+func localInfoPOSTTask(ctx *getconfigContext) {
+
+	log.Functionf("localInfoPOSTTask: waiting for localInfoPOSTTrigger")
+	// wait for the first trigger
+	<-ctx.localInfoPOSTTicker.C
+	log.Functionln("localInfoPOSTTask: waiting for localInfoPOSTTrigger done")
+	// trigger again to pass into the loop
+	triggerLocalInfoPOST(ctx)
+
+	wdName := agentName + "-localinfo"
+
+	// Run a periodic timer so we always update StillRunning
+	stillRunning := time.NewTicker(25 * time.Second)
+	ctx.zedagentCtx.ps.StillRunning(wdName, warningTime, errorTime)
+	ctx.zedagentCtx.ps.RegisterFileWatchdog(wdName)
+
+	for {
+		select {
+		case <-ctx.localInfoPOSTTicker.C:
+			start := time.Now()
+
+			sendLocalInfo(ctx)
+
+			ctx.zedagentCtx.ps.CheckMaxTimeTopic(wdName, "localInfoPOSTTask", start,
+				warningTime, errorTime)
+		case <-stillRunning.C:
+		}
+		ctx.zedagentCtx.ps.StillRunning(wdName, warningTime, errorTime)
+	}
+}
+
+func sendLocalInfo(ctx *getconfigContext) {
+	localProfileServer := ctx.localProfileServer
+	if localProfileServer == "" {
+		return
+	}
+	localServerURL, err := makeLocalServerBaseURL(localProfileServer)
+	if err != nil {
+		log.Errorf("sendLocalInfo: makeLocalServerBaseURL: %v", err)
+		return
+	}
+	if !ctx.localServerMap.upToDate {
+		err := updateLocalServerMap(ctx, localServerURL)
+		if err != nil {
+			log.Errorf("sendLocalInfo: updateLocalServerMap: %v", err)
+			return
+		}
+	}
+	srvMap := ctx.localServerMap.servers
+	if len(srvMap) == 0 {
+		log.Functionf("sendLocalInfo: cannot find any configured apps for localServerURL: %s",
+			localServerURL)
+		return
+	}
+
+	localInfo := prepareLocalInfo(ctx)
+
+	var errList []string
+	for bridgeName, servers := range srvMap {
+		for _, srv := range servers {
+			fullURL := srv.localServerAddr + localInfoURLPath
+			resp, err := zedcloud.SendLocalProto(
+				zedcloudCtx, fullURL, bridgeName, srv.bridgeIP, localInfo, nil)
+			if err != nil {
+				errList = append(errList, fmt.Sprintf("SendLocalProto: %v", err))
+				continue
+			}
+			if resp.StatusCode == http.StatusNotFound {
+				//throttle sending
+				updateLocalInfoTicker(ctx, 5)
+			}
```

Comment:
I can see cases where there is a localprofileserver configured, but it does not implement the /info API endpoint. (It might only implement the profile part, or profile+radiosilence).
In that case it seems quite wasterful to keep on sending this periodically.
We'd need some retry in case the local profile server hasn't started yet, or there is a temporary failure, but for a 404 I think we should not retry.

Reply:
What I try to mitigate here is some kind of moving from existing version to the next one (with implemented info endpoint). In general, if reboot of the node after the new version of server deployed is not a problem, we can stop sending in case of 404.

Reply:
Or changing the local profile server in the config to cause everything to be resent.

Reply:
OK. Can you ping @rvs to check if he thinks his request for changes has been taken care of?